src/main/java/org/folio/service/finance/transaction/EncumbranceRelationsHoldersBuilder.java

```diff
@@ -235,9 +235,11 @@ private EncumbranceRelationsHolder buildBaseHolder(EncumbranceRelationsHolder ho
       List<Transaction> transactionsFromStorage) {
 
     List<EncumbranceRelationsHolder> toBeReleasedHolders = transactionsFromStorage.stream()
-      .filter(transaction -> encumbranceHolders.stream().allMatch(holder -> !transaction.getEncumbrance().getSourcePoLineId().equals(holder.getPoLineId())
+      .filter(transaction -> encumbranceHolders.stream().allMatch(holder ->
+        (holder.getOldEncumbrance() == null || !transaction.getId().equals(holder.getOldEncumbrance().getId())) && (
```

Comment:
Please move all conditions to the separate method for improving code readability : isTransactionToBeReleased()

Reply:
Good idea, these conditions are hard to understand. Also with a little logic change I can use the same match method for withToBeReleasedHolders() and mapHoldersToTransactions().